src/Features/Core/Portable/CodeRefactorings/AddMissingImports/AddMissingImportsAnalysisResult.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.AddImport;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.AddMissingImports
+{
+    internal sealed class AddMissingImportsAnalysisResult
+    {
+        public ImmutableArray<AddImportFixData> AddImportFixData { get; }
+        public Document Document { get; }
+        public TextSpan TextSpan { get; }
+        public bool CanAddMissingImports { get; }
```

Comment:
i don't really get the connection between this item and the top item.  What would it mean to have many datas, but have this be false?  What would it mean to have no data, but have this be true?

What is the TextSpan for?  What is the result Doucment?

Reply:
TextSpan is the area that the import fixes applies to within the document, document is the document the fixes apply to. I'm happy to remove the bool if just checking `AddimportFixData.IsEmpty` is more clear?

Reply:
> I'm happy to remove the bool if just checking AddimportFixData.IsEmpty is more clear?

You can keep the bool.  However, if it means the same thing, then it's better to have as `public bool CanAddMissingImports => !AddImportFixData.IsEmpty`

Reply:
> TextSpan is the area that the import fixes applies to within the document, document is the document the fixes apply to. 

But isn't that hte *input* to the analysis request?  why is it contained in the result?

Reply:
Document is needed since it isn't stored on the AddImportFixData, but the TextSpan can be dropped

Reply:
Continuing in https://github.com/dotnet/roslyn/pull/48501#discussion_r512336345